cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -91,33 +91,27 @@ For these cases, attempting to retrofit this pattern in existing applications re
+The Encrypted Token Pattern leverages an encryption, rather than comparison method of Token-validation. It is most suitable for applications that do not want to maintain any state at server side. 
 
-After successful authentication, the server generates a unique token comprised of the user's ID, a timestamp value and a [nonce](http://en.wikipedia.org/wiki/Cryptographic_nonce), using a unique key available only on the server. This token is returned to the client and embedded in a hidden field for forms, in the request-header/parameter for AJAX requests. On receipt of this request, the server reads and decrypts the token value with the same key used to create the token. Inability to correctly decrypt suggest an intrusion attempt. Once decrypted, the UserId and timestamp contained within the token are validated; the UserId is compared against the currently logged in user, and the timestamp is compared against the current time.
+Server generates a token comprised of the user's session ID and a timestamp (to prevent replay attacks) using a unique key available only on the server (AES256 or stronger encryption algorithms are recommended). This token is returned to the client and embedded in a hidden field for forms, in the request-header/parameter for AJAX requests. On receipt of this request, the server reads and decrypts the token value with the same key used to create the token. Inability to correctly decrypt suggest an intrusion attempt (recommend to block and log the attack for incident response purposes). Once decrypted, the users sessionId and timestamp contained within the token are validated; session-id is compared against the currently logged in user, and the timestamp is compared against the current time to verify that its not beyond the defined token expiry time. If session-id matches and the timestamp is under the defined token expiry time, request can be allowed.
 
-On successful token-decryption, the server has access to parsed values, ideally in the form of [claims](http://en.wikipedia.org/wiki/Claims-based_identity). These claims are processed by comparing the UserId claim to any potentially stored UserId (in a Cookie or Session variable, if the site already contains a means of authentication). The Timestamp is validated against the current time, preventing replay attacks. Alternatively, in the case of a CSRF attack, the server will be unable to decrypt the poisoned token, and can block and log the attack.
-
-This technique addresses some of the shortfalls in other stateless approaches, such as the need to store data in a Cookie, circumnavigating the Cookie-subdomain and HTTPONLY issues.
+If you would like to employ key rotations (recommended), you can include key related information (number representing rotation or any other identifier basing on your organizational/individual key management systems being used) in the token generation process and use that identifier while validating back at the server.
```

Comment:
You can propose a example of representation of the key related information (or a link to an example) to guide the reader to apply your proposal.

What do you think?

Reply:
Yeah, I can now see that the current version is not easy to understand. 

For now, I will re-submit without any information about key rotations (as it was before this PR) and when I get some free time over next week, I will submit a new PR with clear example about key rotation.

Reply:
A small example is enough, it's just to guide a developer to use your recommandation "from scrach" with a simple representation of the key related information in the token. 

It's a kind of getting-started hint...

Reply:
Updated you already on this :)

Reply:
Perhaps i will be wrong but according to the kind of value protected here (CSRF token) we can skip the key rotation topic because it can be overkill according to the crypto usage here. 

It can be interesting to add a link to [this section](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Key_Management_Cheat_Sheet.md#key-management-lifecycle-best-practices) of the Key management CS in order to provide hint about key storage/lifecyle.

What do you think?

Reply:
As this is discussing keys, it doesn't hurt to let the reader know about the key management CS.

Reply:
I like the idea of cross-reference to Key management CS